### PR TITLE
test-modeling-rules-fixes

### PR DESCRIPTION
**Pull request by sapirshuker**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7147

## Description
Fixed an issue where the **modeling-rules test** command failed to get the existence of result from dataset in cases where the results take time to load.

### CODE REVIEW

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 1]
```diff
@@ -287,27 +288,38 @@ def check_dataset_exists(
...
+                    logger.info(
+                        f"[cyan]try to get results from the data set, there are not results for the {i}th time. continue.[/cyan]",
+                        extra={"markup": True},
                     )
-                    logger.error(err, extra={"markup": True})
-                break
             except requests.exceptions.HTTPError:
                 pass
             sleep(interval)
+        # there are no results from the dataset but it exists if the dataset doesn't exists we will catch it in except section.
```
Comment (daryakoval) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:310]:
```suggestion
        # There are no results from the dataset but it exists. If the dataset doesn't exist HTTPError exception is raised.
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 2]
```diff
@@ -287,28 +289,41 @@ def check_dataset_exists(
...
                 else:
-                    err = (
-                        f"[red]Dataset {dataset} exists but no results were returned. This could mean that your testdata "
-                        "does not meet the criteria for an associated Parsing Rule and is therefore being dropped from "
-                        "the dataset. Check to see if a Parsing Rule exists for your dataset and that your testdata "
-                        "meets the criteria for that rule.[/red]"
+                    dataset_exist = True
+                    results_exist = False
+                    logger.info(
+                        f"[cyan]try to get results from the data set, there are not results for the {i+1}th time. continue.[/cyan]",
```
Comment (Shellyber) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:307]:
```suggestion
                        f"[cyan]trying to get results from the dataset for the {i+1}th time. continuing to try to get the results.[/cyan]",
```

File: demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py [hunk 3]
```diff
@@ -287,28 +289,41 @@ def check_dataset_exists(
...
+                        extra={"markup": True},
                     )
-                    logger.error(err, extra={"markup": True})
-                break
             except requests.exceptions.HTTPError:
                 pass
             sleep(interval)
+        # There are no results from the dataset but it exists. If the dataset doesn't exist HTTPError exception is raised.
```
Comment (Shellyber) [inline demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py:313]: The dataset exists here, but no results. The second part of the comment here is wrong.

**Comment by Shellyber** [comment]
@sapirshuker please add here a comment with a build that shows that this fix worked. Also a screenshot

**Comment by sapirshuker** [comment]
@Shellyber  job number is #28378762
Screenshot 2023-07-02 at 18 54 33

**Review by Shellyber — APPROVED** [review]
Nice work

**Pull request merged**